target/i386: add MDS-NO feature

Microarchitectural Data Sampling is a hardware vulnerability which allows
unprivileged speculative access to data which is available in various CPU
internal buffers.

Some Intel processors use the ARCH_CAP_MDS_NO bit in the
IA32_ARCH_CAPABILITIES
MSR to report that they are not vulnerable, make it available to guests.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Message-Id: <20190516185320.28340-1-pbonzini@redhat.com>
Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>

Author: Paolo Bonzini <pbonzini@redhat.com>
Origin: upstream, https://github.com/qemu/qemu/commit/20140a82c674
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1828495
Reviewed-By: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Last-Update: 2019-08-02

--- qemu-2.11+dfsg.orig/target/i386/cpu.c
+++ qemu-2.11+dfsg/target/i386/cpu.c
@@ -597,7 +597,7 @@ static FeatureWordInfo feature_word_info
         .type = MSR_FEATURE_WORD,
         .feat_names = {
             "rdctl-no", "ibrs-all", "rsba", "skip-l1dfl-vmentry",
-            "ssb-no", NULL, NULL, NULL,
+            "ssb-no", "mds-no", NULL, NULL,
             NULL, NULL, NULL, NULL,
             NULL, NULL, NULL, NULL,
             NULL, NULL, NULL, NULL,
